casanovo/casanovo.py

```diff
@@ -489,6 +522,44 @@ def _get_model_weights(cache_dir: Path) -> str:
             )
 
 
+def _resolve_output(
+    output_dir: str | None,
+    output_root: str | None,
+    file_patterns: list[str],
+    verbosity: str,
+) -> Tuple[Path, str]:
+    """
+    Resolves the output directory and sets up logging.
+
+    Parameters:
+    -----------
+    output_dir : str | None
+        The path to the output directory. If `None`, the current working
+        directory will be used.
+    output_root : str | None
+        The base name for the output files. If `None`, no specific base name is
+        set, and logging will be configured accordingly to the behavior of
+        `setup_logging`.
+    file_patterns : list[str]
+        A list of file patterns that should be checked within the `output_dir`.
+    verbosity : str
+        The verbosity level for logging.
+
+    Returns:
+    --------
+    Tuple[Path, str]
+        The output directory and the base name for log and results files (if
+        applicable).
+    """
+    output_dir = Path(output_dir) if output_dir is not None else Path.cwd()
+    output_base_name = (
+        None if output_root is None else (output_dir / output_root)
+    )
+    utils.check_dir(output_dir, file_patterns)
+    output = setup_logging(output_base_name, verbosity)
+    return output, output_dir
+
```

Comment:
It might be better to pop a warning here instead, not sure.

Reply:
I'm not sure what this comment refers to. What do we need to warn about?

Reply:
This comment was supposed to be on the line `raise FileNotFoundError(`, my bad.